### ISSUE TITLE
`IM_ASSERT(tab->ContentWidth > 0.0f)` while using `ImGuiConfigFlags_DpiEnableScaleFonts`

### ISSUE THREAD

**Issue by nicolasnoble** [ocornut/imgui#5572]
This is basically a cross post of https://github.com/grumpycoders/pcsx-redux/issues/979

When using the `docking` branch with the `ImGuiConfigFlags_DpiEnableScaleFonts` config flag, it is possible to get in a situation where the code asserts at runtime.

Here is how to reproduce it with the base demo code. When running `example_glfw_opengl3`, add the line `   io.ConfigFlags |= ImGuiConfigFlags_DpiEnableScaleFonts;` to main.cpp. Then take the two demo windows, assemble them in a docked manner, and move them out of the screen slightly, like so:

![image](https://user-images.githubusercontent.com/7281574/184749948-9de0af92-d461-43ed-b362-586344001bab.png)

It should look like this:

![image](https://user-images.githubusercontent.com/7281574/184750528-81ca7b75-ac23-4b5e-b02d-fcc383daa852.png)

At this point, close and re-open the demo. When re-loading the settings, the following assert should trigger:

https://github.com/ocornut/imgui/blob/205874f5b15a3689d253c6c149c62ad704dae0ab/imgui_widgets.cpp#L7077

(introduced in https://github.com/ocornut/imgui/commit/205874f5b15a3689d253c6c149c62ad704dae0ab)

The reason for this is that with `ImGuiConfigFlags_DpiEnableScaleFonts`, the first frame won't have proper font sizes yet for those windows. It'll be fine after a full run of the main loop, but the first time, when we calculate the tab's content width, `TabItemCalcSize` will return 0, because `g.FontSize` is 0 at this point. This affects both the `CalcTextSize` calculation, which returns a `(0,0)` as a result, and `TabItemCalcSize` itself, which does an `ImMin` of the return value of `CalcTextSize` and `TabBarCalcMaxTabWidth()`, with the latter returning 0 too due to `g.FontSize` being 0 again.

A quick workaround which works locally here with no apparent ill-effect here is to cap the return value of `TabItemCalcSize` by adding an `ImMax(1.0f, ...` the following way:

```c
    return ImVec2(ImMax(1.0f, ImMin(size.x, TabBarCalcMaxTabWidth())), size.y);
```

The frame will be wrong the first time, which is basically a guarantee of using `ImGuiConfigFlags_DpiEnableScaleFonts`, and then everything corrects itself the next frame properly.

Trying to remove the assert itself will cause further problems down the line.

**Comment by ocornut** [comment]
Thank you. I think we should be investigating the source issue which is zero scaling. Will do.

**Comment by ocornut** [comment]
I pushed 3e6f948 which fixes things on the tab size.

However keeping this opened as I believe there's another bug with the dpi scaling creating a zero font size on initial viewport frame.

**Comment by ocornut** [comment]
I wanted to investigate this, namely FontSize being zero for a frame but couldn't repro.

The zero would perhaps come from this factor:
```cpp
window->FontDpiScale = (g.IO.ConfigFlags & ImGuiConfigFlags_DpiEnableScaleFonts) ? window->Viewport->DpiScale : 1.0f;
```
Which should come from:
```cpp
        // Update DPI scale
        float new_dpi_scale;
        if (g.PlatformIO.Platform_GetWindowDpiScale && platform_funcs_available)
            new_dpi_scale = g.PlatformIO.Platform_GetWindowDpiScale(viewport);
        else if (viewport->PlatformMonitor != -1)
            new_dpi_scale = g.PlatformIO.Monitors[viewport->PlatformMonitor].DpiScale;
        else
            new_dpi_scale = (viewport->DpiScale != 0.0f) ? viewport->DpiScale : 1.0f;
        [...]
        viewport->DpiScale = new_dpi_scale;
```
Browsing your repo from August 14 (https://github.com/grumpycoders/pcsx-redux/tree/6d3075cf557b192aafcd17fd1a8434e0d6ac0350)

- Platform_GetWindowDpiScale should be NULL.
- So it should be pulling from `g.PlatformIO.Monitors[viewport->PlatformMonitor].DpiScale` ? Hypothesis: the call to `glfwGetMonitorContentScale()` under `#if GLFW_HAS_PER_MONITOR_DPI` returned 0.0f:
https://github.com/ocornut/imgui/blob/dedb381c510cc0b87164e16b9e7ef6bf50ffccec/backends/imgui_impl_glfw.cpp#L520
https://github.com/ocornut/imgui/blob/dedb381c510cc0b87164e16b9e7ef6bf50ffccec/imgui.h#L3063

Unsure how it could happen and it is possible it happens from another source due to a higher-level construct.

Basically `g.FontSize` should never be 0.0f.
Can you add a `IM_ASSERT(scale > 0.0f)` in `ImGuiWindow::CalcFontSize()` and see if it ever triggers on your setup?
